KubeJS support for the Nuclear Reactor and Fusion Reactor

globalista-dev opened the pull request:
**Todo list**
- [x] Custom fuel rods
- [x] Custom control rods
- [x] Control rod part 
- [x] Custom isotope params
- [x] Custom isotope fuel params
- [x] Add documentation for custom fuel rods and control rods
- [x] Custom fluid nuclear components

Documentation needs further explanations

Technici4n commented on docs/ADDING_NUCLEAR_FUELS.md:
````diff
@@ -0,0 +1,185 @@
+
+```javascript
+MIMaterialEvents.addMaterials(event => {
+
````
I don't like these empty lines after an opening brace.

Technici4n commented on docs/ADDING_NUCLEAR_FUELS.md:
````diff
@@ -0,0 +1,185 @@
+This generates the same parameters as LE MOX fuel
+
+``` javascript
+[...]
````
You should provide full examples to reduce the amount of questions by people not familiar with JS.

Technici4n commented on src/main/java/aztech/modern_industrialization/compat/kubejs/nuclear/CreateIsotopeFuelParamsEventJS.java:
```diff
@@ -0,0 +1,58 @@
+
+    public static SortedMap<String, NuclearConstant.IsotopeFuelParams> FUEL_PARAMS = new TreeMap<>();
+
+    public void createIsotopeFuelParams(String name, double thermalAbsorbProba, double thermalScatterings, int maxTemp, int tempLimitLow,
```
I think that these events are a bit of a roundabout way to go about things. My intuition is that the registration of nuclear components can probably be simplified a little to make this unnecessary? Not sure

globalista-dev replied:
Yeah, I thought about that as well. I couldn't find a way to make it less complicated, but I'll try and come up with a better way to do things

Technici4n commented on src/main/java/aztech/modern_industrialization/nuclear/FluidNuclearComponents.java:
```diff
@@ -51,49 +56,51 @@ public class FluidNuclearComponents {
-        } else if (fluid == MIFluids.HIGH_PRESSURE_WATER.asFluid()) {
-            return HIGH_PRESSURE_WATER;
-        } else if (fluid == MIFluids.HIGH_PRESSURE_HEAVY_WATER.asFluid()) {
-            return HIGH_PRESSURE_HEAVY_WATER;
```
This whole function can be a `get` call at this point. :P

globalista-dev replied:
The _of_ function?

Technici4n replied:
yeah - `get` will return null if the value is not in the map already

globalista-dev replied:
You're right xD
Fixed it

Technici4n reviewed (changes requested):
very promising, just a bit roundabout (but that's the fault of the current system, not your PR)

globalista-dev commented:
> I don't like these empty lines after an opening brace.
Fixed

> You should provide full examples to reduce the amount of questions by people not familiar with JS.
Added full examples to everything

globalista-dev commented:
Moved both params to the same class (RegisterNuclearParams) and to the MIRegistrationKubeJSEvents interface. Both types of params are using the same map, but that shouldn't really be a problem.
